packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,178 @@
+   */
+  link?: {|
+    accessibilityLabel?: string,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    text?: string,
+  |},
+|};
+
+const DEFAULT_ZINDEX = 2;
+
+/**
+ * [HelpButton](https://gestalt.pinterest.systems/web/helpbutton) is an affordance that accompanies an element on the screen. It helps describe or provide assistance on how to use the accompanying element.
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  text,
+  zIndex,
+}: Props): Node {
+  const ref = useRef(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isHover, setIsHover] = useState(false);
+  const {
+    accessibilityTooltipMessage,
+    accessibilityLinkLabel: accessibilityDefaultLinkLabel,
+    accessibilityIcon,
+  } = useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (isOpen && ref.current) {
+      ref.current.focus();
+    }
+  }, [isOpen]);
+
+  const toggleView = () => {
+    setIsOpen((currVal) => !currVal);
+  };
+
+  const bgIconColor = isHover ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(DEFAULT_ZINDEX - 1);
+
+  return (
+    <Fragment>
+      <Box dangerouslySetInlineStyle={{ __style: { display: 'inline-block' } }} tabIndex={-1}>
+        <TapArea
+          accessibilityExpanded={isOpen}
+          accessibilityControls="info-dialog"
+          accessibilityLabel={accessibilityTooltipMessage}
+          fullWidth={false}
+          onTap={toggleView}
+          ref={ref}
+          role="button"
+          rounding="circle"
+          onMouseEnter={() => setIsHover(true)}
+          onMouseLeave={() => setIsHover(false)}
+        >
+          <Tooltip
+            text={accessibilityTooltipMessage}
+            idealDirection={idealDirection}
+            zIndex={tooltipZIndex}
+          >
+            <Box
+              width={16}
+              height={16}
+              rounding="circle"
+              display="flex"
+              alignItems="center"
+              justifyContent="center"
+              dangerouslySetInlineStyle={{
+                __style: {
+                  backgroundColor: `var(--color-text-${bgIconColor})`,
+                },
+              }}
+            >
+              <Icon
+                accessibilityLabel={accessibilityIcon}
+                color="inverse"
+                icon="question-mark"
+                size={8}
+              />
+            </Box>
+          </Tooltip>
+        </TapArea>
+      </Box>
+      {isOpen && (
+        <Layer
+          zIndex={
+            zIndex
+              ? new CompositeZIndex([new FixedZIndex(zIndex.index())])
+              : new FixedZIndex(DEFAULT_ZINDEX)
+          }
+        >
+          <Popover
+            id="info-dialog"
```

Comment:
This needs to be unique on the page. Let's use [useId](https://beta.reactjs.org/reference/react/useId) instead.

Reply:
Which case we should use or not?